graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py

```diff
@@ -0,0 +1,107 @@
+# standard/multi-vendor types have no guarantee of working on all backends, and will only work
+# on those backends that explicitly support them by name. That is, the type will always emit its
+# exact name in DDL with CREATE TABLE is issued."
+# TODO(pmantica1): Add scalar mapping for the following classes: Interval.
+GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE = {
+    sqltypes.BIGINT: GraphQLInt,
+    sqltypes.BigInteger: GraphQLInt,
+    sqltypes.Boolean: GraphQLBoolean,
+    sqltypes.CHAR: GraphQLString,
+    sqltypes.CLOB: GraphQLString,
+    sqltypes.Date: GraphQLDate,
+    sqltypes.DATE: GraphQLDate,
+    sqltypes.DateTime: GraphQLDateTime,
+    sqltypes.DATETIME: GraphQLDateTime,
+    sqltypes.Enum: GraphQLString,
+    sqltypes.Float: GraphQLFloat,
+    sqltypes.FLOAT: GraphQLFloat,
+    sqltypes.INT: GraphQLInt,
+    sqltypes.Integer: GraphQLInt,
+    sqltypes.INTEGER: GraphQLInt,
+    sqltypes.NCHAR: GraphQLString,
+    sqltypes.Numeric: GraphQLDecimal,
+    sqltypes.NUMERIC: GraphQLDecimal,
+    sqltypes.NVARCHAR: GraphQLString,
+    sqltypes.REAL: GraphQLFloat,
+    sqltypes.SMALLINT: GraphQLInt,
+    sqltypes.SmallInteger: GraphQLInt,
+    sqltypes.String: GraphQLString,
+    sqltypes.Text: GraphQLString,
+    sqltypes.TEXT: GraphQLString,
+    sqltypes.Time: GraphQLDateTime,
+    sqltypes.TIME: GraphQLDateTime,
+    sqltypes.TIMESTAMP: GraphQLDateTime,
+    sqltypes.Unicode: GraphQLString,
+    sqltypes.UnicodeText: GraphQLString,
+    sqltypes.VARCHAR: GraphQLString,
+}
+
+# We do not currently plan to add a mapping for JSON and Binary objects.
+UNSUPPORTED_GENERIC_SQL_TYPES = frozenset({
+    sqltypes.ARRAY,
+    sqltypes.Binary,
+    sqltypes.BINARY,
+    sqltypes.Interval,
+    sqltypes.JSON,
+    sqltypes.LargeBinary,
+    sqltypes.PickleType,
+    sqltypes.VARBINARY,
+})
+
+MSSQL_CLASS_TO_GRAPHQL_TYPE = {
+    mssqltypes.BIT: GraphQLBoolean,
+    mssqltypes.DATETIME2: GraphQLDateTime,
+    mssqltypes.REAL: GraphQLFloat,
+    mssqltypes.NTEXT: GraphQLString,
+    mssqltypes.SMALLDATETIME: GraphQLDateTime,
+    mssqltypes.TIME: GraphQLDateTime,
+    mssqltypes.TINYINT: GraphQLInt,
+    mssqltypes.UNIQUEIDENTIFIER: GraphQLID,
+
+}
+
+UNSUPPORTED_MSSQL_TYPES = frozenset({
+    mssqltypes.DATETIMEOFFSET,
+    mssqltypes.IMAGE,
+    mssqltypes.MONEY,
+    mssqltypes.ROWVERSION,
+    mssqltypes.SQL_VARIANT,
+    mssqltypes.SMALLMONEY,
+    # The mssqltypes.TIMESTAMP class inherits from SQL._Binary class, which is not supported.
+    # TIMESTAMP docstring:
+    #     Note this is completely different than the SQL Standard
+    #     TIMESTAMP type, which is not supported by SQL Server.  It
+    #     is a read-only datatype that does not support INSERT of values.
+    mssqltypes.TIMESTAMP,
+    mssqltypes.VARBINARY,
+    mssqltypes.XML,
+})
+
+SQL_CLASS_TO_GRAPHQL_TYPE = {}
+SQL_CLASS_TO_GRAPHQL_TYPE.update(GENERIC_SQL_CLASS_TO_GRAPHQL_TYPE)
+SQL_CLASS_TO_GRAPHQL_TYPE.update(MSSQL_CLASS_TO_GRAPHQL_TYPE)
```

Comment:
https://github.com/kensho-technologies/graphql-compiler/blob/macro_system/graphql_compiler/macros/macro_expansion.py#L21 Maybe we should make a common util function for this.

Reply:
Yea makes sense. I'll create a global_utils file in the top level of the directory.

Reply:
Actually just noticed that you have this in the macro branch. I would have to make like 2 extra PRs to add this. (Add it into master and remove the function from macro_system). This seems like too much work for this. I'll add a TODO.

Reply:
Actually just noticed that you have this in the macro branch. I would have to make like 2 extra PRs to add this. (Add it into master and remove the function from macro_system). This seems like too much work for this. I'll add a TODO to use it when it gets merged into master.